pkg/api/service.go

```diff
@@ -104,7 +105,19 @@ func (s *ServiceSpec) Validate() error {
 		return fmt.Errorf("invalid mode: %q", s.Mode)
 	}
 
-	// TODO: validate the service name is a valid DNS label.
+func (s *ServiceSpec) Validate() error {
```

Comment:
Please check that you've duplicated the `Validate` function so that there are syntax errors in the file and the code doesn't compile.

Reply:
It would also be great to add a few basic unit tests for `Validate` with valid and invalid service name. You can create a `service_test.go` file at `pkg/api` for that.

Please also make sure that all existing tests also pass by running at the root of the project: `go test ./...`
Please make sure you have Docker installed locally as it will also run e2e tests using a local uncloud cluster in Docker containers.